cpp/tests/spatial/hausdorff_compare.cu

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
What's the point of renaming it from `hausdorff_test` to `hausdorff_compare`? I think `hausdorff_test was a perfectly good name.

Reply:
*_test.cu is now for test cuSpatial only, using real data.  Some many execute twice to test consistency. 
*_compare.cu now compares CPU and GPU results using the same real data. 
These are two separate tests. Some tests may not have compare code.

Reply:
Note that multiple tests of the same module can go in the same source file. See libcudf for suggested test organization.

Reply:
Please feel free to merge. I am still far from a gtest expert :-)
On the other hand, having _toy,_test,_compare separate might enable build time re-configuration, which could be a desirable feature than using file level marcos.